nucleus-sync/src/integration-test/java/org/alfresco/hxi_connector/nucleus_sync/client/NucleusClientIntegrationTest.java

```diff
@@ -0,0 +1,357 @@
+                  ]
+                }
+                """;
+
+        wireMockServer.stubFor(get(urlEqualTo("/api/users"))
+                .withHeader("Authorization", equalTo("Bearer nucleus-token"))
+                .willReturn(aResponse()
+                        .withStatus(200)
+                        .withHeader("Content-Type", "application/json")
+                        .withBody(responseBody)));
+
+        // Act
+        List<IamUser> users = nucleusClient.getAllIamUsers();
+
+        // Assert
+        assertThat(users).extracting(IamUser::getUserId)
+                .containsExactlyInAnyOrder("714e1cbd-d88a-4fd1-a491-be438f7a2233",
+                        "34f8e319-6a8f-4359-a5b1-d26a04a4abc0");
+        assertThat(users).extracting(IamUser::getUserName)
+                .containsExactlyInAnyOrder("Jane.Doe+cin@hyland.com", "moliver");
+        assertThat(users).extracting(IamUser::getEmail)
+                .containsExactlyInAnyOrder("Jane.Doe+cin@hyland.com", "Michale.Oliver+cin@hyland.com");
+    }
+
+    @Test
+    void testGetAllExternalGroups_Success()
+    {
+        // Arrange
+        String responseBody = """
+                {
+                  "items": [
+                    {
+                      "externalGroupId": "GROUP_HR",
+                      "attributes": []
+                    },
+                    {
+                      "externalGroupId": "GROUP_MARKETING",
+                      "attributes": []
+                    }
+                  ]
+                }
+                """;
+
+        wireMockServer.stubFor(get(urlEqualTo("/system-integrations/systems/" + SYSTEM_ID + "/groups"))
+                .withHeader("Authorization", equalTo("Bearer nucleus-token"))
+                .willReturn(aResponse()
+                        .withStatus(200)
+                        .withHeader("Content-Type", "application/json")
+                        .withBody(responseBody)));
+
+        // Act
+        List<NucleusGroupOutput> groups = nucleusClient.getAllExternalGroups();
+
+        // Assert
+        assertThat(groups).extracting(NucleusGroupOutput::getExternalGroupId)
+                .containsExactlyInAnyOrder("GROUP_HR", "GROUP_MARKETING");
+    }
+
+    @Test
+    void testGetCurrentUserMappings_Success()
+    {
+        // Arrange
+        String responseBody = """
+                {
+                  "items": [
+                    {
+                      "userId": "18a17e9d-dbb1-4643-a2e7-3e1859961f5b",
+                      "externalUserId": "jdoe",
+                      "attributes": []
+                    },
+                    {
+                      "userId": "24bd547d-58b3-4722-889a-84e68c41615b",
+                      "externalUserId": "moliver",
+                      "attributes": []
+                    }
+                  ]
+                }
+                """;
+
+        wireMockServer.stubFor(get(urlEqualTo("/system-integrations/systems/" + SYSTEM_ID + "/user-mappings"))
```

Comment:
Can a user have more than one external user id in a single external system? If so then we should test this.